### PR TITLE
Convert CalendarContainer to Hooks and add a better webpack config for handling aliases

### PR DESCRIPTION
**Pull request by martiuh**
CHANGES

- CalendarContainer.js from class component to function component (with hooks)
- created a `config/customWebpackConfig.js` to better handling custom configuration, such as creating alias for both production and development.
- `src/components/MaterialButton/MaterialButton.js` better name desctructuring
- `src/components/NavBar/UsernameLabel.jsx` drop `lodash/capitalize` (It does not capitalize two word names)
- Created the `context` folder, where all the contexts used in the app should be listed (For easier use with Hooks)
- Rename `src/providers/Auth` to `src/providers/AuthProvider`

### CODE REVIEW

File: src/pages/Calendar/Calendar.jsx [hunk 1]
```diff
@@ -8,10 +8,12 @@ import { BookingsSideBar } from "components/BookingsSideBar/BookingsSideBar.jsx"
 import { withAuthContext } from "hocs";
 import { NotificationProvider, ModalFormProvider } from "providers";
 import { NoteCard } from "components/NoteCard";
+import { AuthContext } from "context/AuthContext";
 
-const CalendarPage = props => {
+export function Calendar({ bookingsData, onBookingsDataChange }) {
```
Comment (mishelashala) [inline src/pages/Calendar/Calendar.jsx]: favor function expressions over function declarations

File: src/pages/Room/CalendarContainer.jsx [hunk 2]
```diff
@@ -1,91 +1,82 @@
...
 
-class CalendarContainerComponent extends React.Component {
-  state = {
-    bookingsData: [],
-    allBookingsData: [],
-    isServerDown: false,
-    roomId: "",
-    isLoading: true
-  };
+function CalendarContainerComponent({ URLRoomId, history }) {
```
Comment (mishelashala) [inline src/pages/Room/CalendarContainer.jsx]: favor function expressions over function declarations

File: src/pages/Room/CalendarContainer.jsx [hunk 3]
```diff
@@ -1,91 +1,82 @@
...
+  const [bookingsData, setBookingsData] = useState([]);
```
Comment (mishelashala) [inline src/pages/Room/CalendarContainer.jsx]: favor `update[State]` over `set[State]`:

```js
const [bookingData, updateBookingData] = useState([]);

File: src/pages/Room/CalendarContainer.jsx [hunk 4]
```diff
@@ -1,91 +1,82 @@
...
-      isLoading
-    } = this.state;
+  if (isServerDown) {
+    return <Error500 />;
+  }
 
-    if (isServerDown) {
-      return <Error500 />;
-    }
+  const onBookingsDataChange = () => setShouldFetch(true);
```
Comment (mishelashala) [inline src/pages/Room/CalendarContainer.jsx]: once it is setted to true there's no way to fetch again, do you think is there any other way to do this?

Comment (martiuh) [reply src/pages/Room/CalendarContainer.jsx]: Sure, I missed that one.

**Pull request merged**